CRM/Case/Task.php

```diff
@@ -60,38 +51,38 @@ class CRM_Case_Task {
    * @return array
    *   the set of tasks for a group of contacts
    */
-  public static function &tasks() {
+  public static function tasks() {
     if (!self::$_tasks) {
       self::$_tasks = array(
-        1 => array(
+        self::TASK_DELETE => array(
```

Comment:
@mattwire @eileenmcnaughton @colemanw - Doesn't this break anything that listens to [hook_civicrm_searchTasks](https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/Hook.php#L904) by virtue of changing the numbers?

Don't get me wrong... the old values are magic-numbers that don't make sense. They seem like a design-flaw. Lots of extensions use the idiom `$tasks[] = array(...)`, which is more robust. 

But... the example in the docs specifically uses the technique:

https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_searchTasks/#example

And there are multiple examples in universe where `hook_civicrm_searchTasks` references numerical IDs/existing constants (`biz.jmaconsulting.bugp/bugp.php`, `biz.jmaconsulting.printgrantpdfs/printgrantpdfs.php`, `civicrm-drupal-6.x/civitest.module.sample`, `uk.co.compucorp.civicrm.giftaid/civigiftaid.php`). Anecdotally, I remember needing to do similar things for... as long as the hook has existed.

(Note: This isn't a pointed critique of #11759 -- it generally applies to all the offspring of #11240.)

Reply:
@totten Yes there is a chance it could break some extensions.  However, the numeric indexes have been a bit inconsistent throughout versions and shouldn't really be relied on - once all the "Task" PRs are merged we'll have a consistent set of constants that extension authors can use, meaning that their extensions won't break in the future!   If we can get them all merged before the 5.0.0 release we can have something in the release notes making extensions authors aware that they may need to tweak the searchTask constants if using hardcoded values in their extensions.

Reply:
As an aside of the ones linked

https://github.com/JMAConsulting/biz.jmaconsulting.bugp/blob/master/bugp.php only adds, as does https://github.com/JMAConsulting/biz.jmaconsulting.printgrantpdfs/blob/master/printgrantpdfs.php as does https://github.com/compucorp/uk.co.compucorp.civicrm.giftaid/blob/master/civigiftaid.php

the test one should go